### PR TITLE
add allowed owner to scaffolder plugin (fix #8205)

### LINKED ISSUE #8205: Scaffolder : specify allowed owner from applications templates
## Scaffolder : Specify allowed owners from RepoUrlPicker

It could be interesting to me to force the organization of the repository for a template. In my case I need all the future applications to be stored inside a specific organization on Github.

## Possible Implementation

On the scaffolder plugin we could add the allowed owners like the allowed host in the RepoUrlPicker.tsx  

## Context

In this way I could parse all the repository inside this specific organization to build CI / CD pipeline

### PR DESCRIPTION
**Pull request by LucileDesfour**

### CODE REVIEW

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx [hunk 1]
```diff
@@ -39,7 +39,10 @@ function splitFormData(url: string | undefined) {
     if (url) {
       const parsed = new URL(`https://${url}`);
       host = parsed.host;
-      owner = parsed.searchParams.get('owner') || undefined;
+      owner =
+        parsed.searchParams.get('owner') || allowedOwners
```
Comment (benjdlambert) [inline plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx:43]: Do we really want this behaviour to default to the first of the `allowedOwners`?

**Comment by benjdlambert** [comment]
This will also need a changeset for the `@backstage/plugin-scaffolder` package. Check the comment for more information.

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx (hunk 1 above)
Comment (LucileDesfour) [reply plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx:43]: I did it cause this is what is done with allowedHost, you think it should be different ?

**Comment by benjdlambert** [comment]
Ok - I think that this is a reasonable change for now, just wondering though that with the work that is going on in #7684 then this might actually change to be a dropdown of all the orgs that you as a signed in user is a part of. Maybe it will be a separate component entirely though, not sure yet of the details.

**Review by benjdlambert — APPROVED** [review]

**Comment by suuus** [comment]
Hey @LucileDesfour, congrats on your first merged PR 🎉 Nice work 👍 Please let me know if we can be helpful in any way.

**Pull request merged**